components/match2/wikis/overwatch/match_summary.lua

```diff
@@ -0,0 +1,195 @@
+
+	local matchSummary = MatchSummary():init()
+
+	matchSummary:header(CustomMatchSummary._createHeader(match))
+		:body(CustomMatchSummary._createBody(match))
+
+	-- comment
+	if match.comment then
+		local comment = MatchSummary.Comment():content(match.comment)
+		matchSummary:comment(comment)
+	end
+
+	-- footer
+	local vods = {}
+	for index, game in ipairs(match.games) do
+		if game.vod then
+			vods[index] = game.vod
+		end
+	end
+
+	match.links.lrthread = match.lrthread
+	match.links.vod = match.vod
+	if not Table.isEmpty(vods) or not Table.isEmpty(match.links) then
+		local footer = MatchSummary.Footer()
+
+		-- Game Vods
+		for index, vod in pairs(vods) do
+			footer:addElement(VodLink.display{
+				gamenum = index,
+				vod = vod,
+				source = vod.url
+			})
+		end
+
+		-- Match Vod + other links
+		local buildLink = function (linkType, link)
+			local linkData = _LINK_DATA[linkType]
+			if not linkData then
+				mw.log('linkType "' .. linkType .. '" is not supported by Module:MatchSummary')
+			else
+				return '[[' .. linkData.icon .. '|link=' .. link .. '|15px|' .. linkData.text .. ']]'
+			end
+		end
+
+		for linkType, link in pairs(match.links) do
+			footer:addElement(buildLink(linkType,link))
+		end
+
+		matchSummary:footer(footer)
+	end
+
+	return matchSummary:create()
+end
+
+function CustomMatchSummary._createHeader(match)
+	local header = MatchSummary.Header()
+
+	header:leftOpponent(header:createOpponent(match.opponents[1], 'left'))
+	      :leftScore(header:createScore(match.opponents[1]))
+	      :rightScore(header:createScore(match.opponents[2]))
+	      :rightOpponent(header:createOpponent(match.opponents[2], 'right'))
+
+	return header
+end
+
+function CustomMatchSummary._createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or (match.date ~= _EPOCH_TIME_EXTENDED and match.date ~= _EPOCH_TIME) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for _, game in ipairs(match.games) do
+		if game.map then
+			body:addRow(CustomMatchSummary._createMapRow(game))
+		end
+	end
+
+	return body
+end
+
+function CustomMatchSummary._gameScore(game, opponentIndex)
+	local score = game.scores[opponentIndex] or ''
+	return htmlCreate('div'):wikitext(score)
+end
+
+function CustomMatchSummary._createMapRow(game)
+	local row = MatchSummary.Row()
+
+	-- Add Header
+	if Logic.isNotEmpty(game.header) then
+		local mapHeader = htmlCreate('div')
+			:wikitext(game.header)
+			:css('font-weight','bold')
+			:css('font-size','85%')
+			:css('margin','auto')
+		row:addElement(mapHeader)
+		row:addElement(MatchSummary.Break():create())
+	end
+
+	local centerNode = htmlCreate('div')
+		:addClass('brkts-popup-spaced')
+		:wikitext(game.mode .. '[[' .. game.map .. ']]')
```

Comment:
Technically it should be an icon but yea there should probably be a space there as well

Reply:
So the module is not ready (as it doesn't fetch an image)?

Reply:
Just the Icon displaying instead of the mode and the team names not displaying properly. Those are the only 2 issues left on the [testing page](https://liquipedia.net/overwatch/User:Fenrir.SPAZ/Sandbox) still to fix

Reply:
Will you be fixing it as part of this PR? Mark as draft if so

Reply:
```suggestion
		-- TODO: Add Game Mode icon
		:wikitext('[[' .. game.map .. ']]')
```